game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProTerritoryValueUtils.java

```diff
@@ -443,16 +444,25 @@ static Collection<Territory> findNearbyEnemyCapitalsAndFactories(
     new BreadthFirstSearch(startTerritory)
         .traverse(
             new BreadthFirstSearch.Visitor() {
+              int currentDistance = -1;
+
               @Override
-              public void visit(final Territory territory) {
+              public boolean visit(Territory territory, int distance) {
                 if (enemyCapitalsAndFactories.contains(territory)) {
                   found.add(territory);
                 }
+                if (distance != currentDistance) {
+                  currentDistance = distance;
+                  // When we reach a new distance, check if we should end the search.
+                  if (!shouldContinueSearch()) {
```

Comment:
Is there something that would speak against passing `distance` as an argument directly and dropping the `currentDistance` field completely, or is there something special about the `found` collection?

Reply:
To preserve behavior, we only check on the transition to the next distance.